rel/value_set_generic.go

```diff
@@ -70,6 +70,17 @@ func NewSet(values ...Value) Set {
 	return set
 }
 
+func CanonicalSet(s Set) Set {
+	if s, ok := s.(GenericSet); ok {
+		values := make([]Value, 0, s.Count())
+		for e := s.Enumerator(); e.MoveNext(); {
+			values = append(values, e.Current())
+		}
+		return NewSet(values...)
+	}
+	return s
+}
+
```

Comment:
I don't get how this fixes the enumerating offset string bug, is this a different feature?

Reply:
It's related. The fix is in the enumerator logic.

Reply:
yep I saw the fix (I'm assuming it's the `Current` function), but how is `CanonicalSet` related? It looks to me it's just taking value from a `GenericSet` and make another `Set`.